src/middlewared/middlewared/plugins/boot.py

```diff
@@ -9,21 +9,29 @@
 from bsd import geom
 
 
+BOOT_POOL_NAME = None
+BOOT_POOL_NAME_VALID = ['freenas-boot', 'boot-pool']
+
+
 class BootService(Service):
 
+    @private
+    async def pool_name(self):
+        return BOOT_POOL_NAME
```

Comment:
Can we at least throw an exception here if `BOOT_POOL_NAME` is None then to prevent unexpected things from happening?

Reply:
Also no, I want to be able to run middlewared outside a FreeNAS install

Reply:
This exception won't prevent you from running middleware outside FN, but will prevent you from running boot pool related tasks, no?

Reply:
boot.pool_name is called on other places as well, not only boot pool tasks, e.g. system dataset

Reply:
And lets face it, the change of this being None in a real FreeNAS install is 0.0000000001% : )